include/deal.II/fe/fe_tools_interpolate.templates.h

```diff
@@ -666,11 +655,19 @@ namespace FETools
       if ((cell->subdomain_id() == subdomain_id) ||
           (subdomain_id == numbers::invalid_subdomain_id))
         {
+#ifdef DEBUG
+          // For continuous elements on grids with hanging nodes we need
+          // hanging node constraints. Consequently, when the elements are
+          // continuous no hanging node constraints are allowed.
+          const bool hanging_nodes_not_allowed =
+            (dof1.get_fe().dofs_per_vertex != 0) || (fe2.dofs_per_vertex != 0);
```

Comment:
I guess it's fine setting this flag for every iteration anew. First, it's only in `Debug` mode and second, it is so much cheaper than everything else we are doing here.

Reply:
That was my thinking. The variable is unused outside the assertion, so I figured that it's worth moving it under the `#ifdef`.